lib/services/local/access_list.go

```diff
@@ -387,18 +408,28 @@ func (a *AccessListService) UpsertAccessListWithMembers(ctx context.Context, acc
 					return trace.Wrap(err)
 				}
 
-				for _, member := range members {
-					// If the member is not in the members map (request), delete it.
-					if existingMember, ok := membersMap[member.GetName()]; !ok {
-						err = a.memberService.WithPrefix(accessList.GetName()).DeleteResource(ctx, member.GetName())
+				for _, existingMember := range members {
+					// If the member is not in the new members map (request), delete it.
+					if newMember, ok := membersMap[existingMember.GetName()]; !ok {
+						err = a.memberService.WithPrefix(accessList.GetName()).DeleteResource(ctx, existingMember.GetName())
 						if err != nil {
 							return trace.Wrap(err)
 						}
 					} else {
+						if !existingMember.Spec.Expires.IsZero() {
+							newMember.Spec.Expires = existingMember.Spec.Expires
+						}
+
+						if existingMember.Spec.Reason != "" {
+							newMember.Spec.Reason = existingMember.Spec.Reason
+						}
+
+						newMember.Spec.AddedBy = existingMember.Spec.AddedBy
```

Comment:
The opposite - it will _prevent_ the `AccessListMember` record from changing if only these values are different to the original. It should stop Member records flagging back and forth between owners during SCIM pushes and/or Sync updates. 

Will add commentary to that effect.

Reply:
Note: I have changes the SCIM group push just to call `UpsertAccessListWithMembers()` rather than do its own synchronisation. So whatever happens _here_ will be the behaviour triggered by a SCIM group push.